### PR TITLE
Improve error handling of Step controller

### PR DESCRIPTION
**Pull request by Martchus**
* Improve error handling of Step controller and clean up the code a little bit.
* See https://progress.opensuse.org/issues/14904

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/Step.pm [hunk 1]
```diff
@@ -49,34 +48,27 @@ sub check_tabmode {
...
+    my $tabmode       = 'screenshot';                 # default
+    my $module_detail = $details->[$testindex - 1];
+    if ($module_detail->{audio}) {
+        $tabmode = 'audio';
+    }
+    elsif ($module_detail->{text}) {
+        my $file = path($job->result_dir(), $module_detail->{text})->open('<:encoding(UTF-8)');
+        my @file_content;
+        if (defined $file) {
+            @file_content = <$file>;
```
[the remark this answers is not in the document]
Comment (kraih) [reply lib/OpenQA/WebAPI/Controller/Step.pm]: Note that `Mojo::File::open` already throws an exception if opening the file failed, so the `defined $file` check shouldn't be doing anything there.

[the remark this answers is not in the document]
Comment (Martchus) [reply lib/OpenQA/WebAPI/Controller/Step.pm]: That block was already there. It just has been re-intended to keep the code simpler. But I can clean it up in a further commit of course.

[the remark this answers is not in the document]
Comment (kraih) [reply lib/OpenQA/WebAPI/Controller/Step.pm]: You re-indent it you own it (in git blame). 😉

[the remark this answers is not in the document]
Comment (foursixnine) [reply lib/OpenQA/WebAPI/Controller/Step.pm]: We actually have a lot of places where we could use slurp and spurt too... but this ain the place to address that :)

[the remark this answers is not in the document]
Comment (Martchus) [reply lib/OpenQA/WebAPI/Controller/Step.pm]: fixed

**Comment by Martchus** [comment]
Seems like using `slurp` breaks one of the tests because now a newline is missing.

**Comment by Martchus** [comment]
In fact the previous code wrongly inserted a white-space and also the test wrongly expected that to be there. So using `slurp` actually fixes a bug and I now adapted the test.

**Pull request merged**